From c67434946744be9573b8b43f87960ad67a5a52bb Mon Sep 17 00:00:00 2001
From: Venkateshwar Rao Gannavarapu <venkateshwar.rao.gannavarapu@xilinx.com>
Date: Tue, 16 Apr 2019 13:40:17 -0700
Subject: [PATCH 0626/1851] dt-bindings: display: xlnx: mixer: Updated
 compatible string

commit 287634812922d22ec30413af1736f10dafb86595 from
https://github.com/Xilinx/linux-xlnx.git

This patch introduces the compatible string for Mixer4.0.

Signed-off-by: Venkateshwar Rao Gannavarapu <venkateshwar.rao.gannavarapu@xilinx.com>
Signed-off-by: Saurabh Sengar <saurabhs@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 .../bindings/display/xlnx/xlnx,mixer.txt           | 14 +++++++++-----
 1 file changed, 9 insertions(+), 5 deletions(-)

diff --git a/Documentation/devicetree/bindings/display/xlnx/xlnx,mixer.txt b/Documentation/devicetree/bindings/display/xlnx/xlnx,mixer.txt
index 635576bf0a46..20f4cec27175 100644
--- a/Documentation/devicetree/bindings/display/xlnx/xlnx,mixer.txt
+++ b/Documentation/devicetree/bindings/display/xlnx/xlnx,mixer.txt
@@ -1,14 +1,16 @@
-Device-Tree bindings for Xilinx Video Mixer IP 3.0 core
+Device-Tree bindings for Xilinx Video Mixer IP core
 
 The IP core provides a flexible video processing block for alpha blending
 and compositing multiple video and/or graphics layers.
-Support for up to eight layers, with an optional logo layer, using a
-combination of video inputs from either frame buffer or streaming video
-cores (through AXI4-Stream interfaces) is provided.
+Support for up to sixteen layers based on IP version, with an optional logo
+layer, using a combination of video inputs from either frame buffer or
+streaming video cores (through AXI4-Stream interfaces) is provided.
 The Video Mixer always has one streaming input layer, known as master layer.
 
 Required properties:
- - compatible: Should be "xlnx,mixer-3.0".
+ - compatible: Must contain atleast one of
+	"xlnx,mixer-4.0" (MIXER 4.0 version)
+	"xlnx,mixer-3.0" (MIXER 3.0 version)
  - reg: Base address and size of the IP core.
  - interrupts: Interrupt number.
  - interrupts-parent: phandle for interrupt controller.
@@ -17,6 +19,8 @@ Required properties:
  - xlnx,bpc: bits per component for mixer
  - xlnx,ppc: pixel per clock for mixer
  - xlnx,num-layers: Total number of layers (excluding logo)
+   Value ranges from 1-9 for compatible string xlnx,mixer-3.0 and
+   Value ranges from 1-17 for comptaible string xlnx,mixer-4.0
  - layer_[x]: node for [x] layer
  - xlnx,layer-id: layer identifier number
  - xlnx,vformat: video format for layer. See list of supported formats below.
-- 
2.31.1

